sql/src/main/java/io/crate/operation/collect/sources/InformationSchemaIterables.java

```diff
@@ -149,6 +158,64 @@ private void createMetaDataBasedIterables(MetaData metaData) {
         ingestionRules = new IngestionRuleInfos(metaData.custom(IngestRulesMetaData.TYPE));
     }
 
+    /**
+     * Iterable for extracting not null constraints from table info.
+     */
+    static class NotNullConstraintIterable implements Iterable<ConstraintInfo> {
+
+        private final TableInfo ti;
+
+        NotNullConstraintIterable(TableInfo ti) {
+            this.ti = ti;
+        }
+
+        @Override
+        public Iterator<ConstraintInfo> iterator() {
+            return new NotNullConstraintIterator(ti);
+        }
+    }
+
+    /**
+     * Iterator that returns ConstraintInfo for each TableInfo with not null constraints.
+     */
+    static class NotNullConstraintIterator implements Iterator<ConstraintInfo> {
+        private final TableInfo tableInfo;
+        private final Iterator<Reference> nullableColumns;
+
+        NotNullConstraintIterator(TableInfo tableInfo) {
+            this.tableInfo = tableInfo;
+            nullableColumns = StreamSupport.stream(tableInfo.spliterator(), false)
+                .filter(reference -> !reference.ident().columnIdent().isSystemColumn()
+                                     && reference.valueType() != DataTypes.NOT_SUPPORTED
+                                     && !reference.isNullable()).iterator();
```

Comment:
maybe just a personal preference, but these conditions are hard to comprehend. Maybe rewriting the negative ones as `== false` is a bit cleaner?
ie. ```reference -> reference.ident().columnIdent().isSystemColumn() == false
                                 && reference.valueType() != DataTypes.NOT_SUPPORTED
                                 && reference.isNullable()).iterator() == false```